server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectWriteAheadLogs.java

```diff
@@ -216,7 +217,7 @@ public void collect(GCStatus status) {
       } finally {
         span5.end();
       }
-
+      hasCollected.set(true);
```

Comment:
You could also wrap this in an:

```java
  if (hasCollected.compareAndSet(false, true)) {
    // stuff in here can only be executed once (so long as it never flips back to false)
    // could use the return value in a Precondition or other check if you really want an error message instead
  }
```

Reply:
This is what I was thinking. I think this approach is best because if we set it only afterwards, there could be a race condition potentially where this block is entered again before the boolean is set at the end of the first run.